operator-common/src/main/java/io/strimzi/operator/common/model/AbstractAuthIdentity.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.model;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.operator.common.Util;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+
+/**
+ * Abstract class to represent the identity used during TLS client authentication
+ */
+public abstract class AbstractAuthIdentity {
```

Comment:
This class declares no member methods (just `static` methods). This suggests to me that it's either not finished, or it's a weak abstraction (whatever behaviour it's abstracting cannot be expressed by any member methods).

- If you just need some functions then you can either put them on appropriate other classes or declare a non-`abstract` class with a private constructor to hold them (no need for `abstract`).
- Or, if you really think there's a useful abstraction here ask yourself why there are no member methods. Seemingly there are no call sites which need to manipulate just a `AbstractAuthIdentity`, so do you actually need an abstraction?